frontend_integration_library.md

```diff
@@ -0,0 +1,154 @@
+
+**\*** sometimes customers might log out, use different devices, browsers or delete their cookies. Every time one of this happens Metrilo will create a new anonymous visitor with a new unique identifier. However, once the visitor identifies himself (enters his email and you call metrilo.identify) Metrilo will check for the existence of a customer with the entered email and if there is any, it will merge him with the anonymous visitor.
+
+```javascript
+  var visitorEmail = 'johnnybravo@gmail.com'; // Visitor's email address.
+  metrilo.identify(visitorEmail);
+```
+
+### Tracking events
+The tracking events which the tracking library supports are:
+* [search](#search)
+* [viewPage](#viewPage)
+* [viewArticle](#viewArticle)
+* [viewCategory](#viewCategory)
+* [viewProduct](#viewProduct)
+* [addToCart](#addToCart)
+* [removeFromCart](#removeFromCart)
+* [checkout](#checkout)
+* [applyTags](#applyTags)
+* [customEvent](#customEvent)
+
+Some of the event functions accept `params`, which is a hash with *optional* parameters. The other parameters (besides `params`) are **required**.
+
+#### search ####
+
+Visitor has searched in the shop (this supports only a plain query string without further filtering). This event was not available in the old tracking library.
+```javascript
+  var query = 'a plant'; // Search query string that the visitor entered in the search form.
+  var url = 'http://myshop.com/search/?s=a+plant'; // The URL that the visitor landed on when submitting the search form.
+
+  metrilo.search(query, url);
+```
+
+#### viewPage ####
+Visitor has visited a shop page.
+
+```javascript
+  var url = 'http://myshop.com/amazing-page' // Absolute URL linking the the visited page.
+  var params = {
+    name: 'My Amazing page' // The name of the page in the shop.
+  };
+
+  metrilo.viewPage(url, params);
+```
+
+#### viewArticle ####
+Visitor has landed on an article page.
+
+```javascript
+  var articleId = '42'; // Unique ID of article resource in the shop's database.
+  var params = {
+    name: 'How to increase sales' // Article display name in the shop.
+    url: 'http://myshop.com/how-to-increase-sales' // Landing page for the article resource.
+  };
+
+  metrilo.viewArticle(articleId, params);
+```
+
+#### viewCategory ####
+Visitor has opened a category page.
```

Comment:
- visited
- landed
- opened
![](https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/06c776e7-09e4-4baa-b4c9-69c03c3856af/d5r4iwa-e5c13fd8-2531-491f-81a1-4f4529ec664e.png/v1/fill/w_900,h_607,q_80,strp/choose_your_weapon__by_firstawesomeplatoon_d5r4iwa-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7ImhlaWdodCI6Ijw9NjA3IiwicGF0aCI6IlwvZlwvMDZjNzc2ZTctMDllNC00YmFhLWI0YzktNjljMDNjMzg1NmFmXC9kNXI0aXdhLWU1YzEzZmQ4LTI1MzEtNDkxZi04MWExLTRmNDUyOWVjNjY0ZS5wbmciLCJ3aWR0aCI6Ijw9OTAwIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmltYWdlLm9wZXJhdGlvbnMiXX0.Z_UCmtQCGea0BxvjzqLwwxC9-uCotH5pEE2SvQ57gSs)

Reply:
I just renamed the file name 😭